source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -54,9 +55,20 @@ Network::FilterStatus ProxyFilter::onNewConnection() {
     return Network::FilterStatus::StopIteration;
   }
 
-  uint32_t default_port = config_->port();
+  const StreamInfo::UInt32Accessor* dynamic_port_filter_state =
+      read_callbacks_->connection()
+          .streamInfo()
+          .filterState()
+          ->getDataReadOnly<StreamInfo::UInt32Accessor>("envoy.upstream.dynamic_port");
+
+  uint32_t port;
+  if (dynamic_port_filter_state) {
+    port = dynamic_port_filter_state->value();
```

Comment:
check whether the port is at right range? for config this is done via PGV.

Reply:
Yeah good catch, updated the code to use the default configured port if the port is out of range, and updated the docs accordingly